expr/literals.go

```diff
@@ -625,6 +677,69 @@ func (t *ProtoLiteral) ToProtoFuncArg() *proto.FunctionArgument {
 func (t *ProtoLiteral) Visit(VisitFunc) Expression { return t }
 func (*ProtoLiteral) IsScalar() bool               { return true }
 
+func NewDecimalWithType(literal *ProtoLiteral, decType *types.DecimalType) (Literal, error) {
+	litType, ok := literal.GetType().(*types.DecimalType)
+	if !ok {
+		return nil, fmt.Errorf("literal type is not decimal")
+	}
+	inDecimalBytes := [16]byte(literal.Value.([]byte))
+	decimalBytes, precision, scale, err := modifyDecimalPrecisionAndScale(inDecimalBytes, litType.Scale, decType.Precision, decType.Scale)
+	if err != nil {
+		return nil, err
+	}
+	return NewLiteral[*types.Decimal](&types.Decimal{Value: decimalBytes[:16], Precision: precision, Scale: scale}, decType.GetNullability() == types.NullabilityNullable)
+}
+
+func NewVarCharWithType(literal *ProtoLiteral, fixedType *types.VarCharType) (Literal, error) {
+	if _, ok := literal.GetType().(*types.VarCharType); !ok {
+		return nil, fmt.Errorf("literal type is not varchar")
+	}
+	if len(literal.Value.(string)) > int(fixedType.GetLength()) {
+		return nil, fmt.Errorf("varchar litearl value length is greater than type length")
+	}
+	return &ProtoLiteral{Value: literal.Value, Type: fixedType}, nil
+}
+
+func NewPrecisionTimestampWithType(literal *ProtoLiteral, ptsType *types.PrecisionTimestampType) (Literal, error) {
```

Comment:
This feels difficult to use as one needs to construct a type object (which isn't easy either) in order to use it.  Since this is a convenience function perhaps just provide the parameter instead?

Reply:
These are convenience functions, current usage has the type available with the caller. Made them local to the module.